src/mscorlib/src/System/Threading/Tasks/Task.cs

```diff
@@ -6191,18 +6191,6 @@ internal static Delegate[] GetDelegatesFromContinuationObject(object continuatio
 
             return null;
         }
-
-        private static Task GetActiveTaskFromId(int taskId)
```

Comment:
@mikem8361 Could you please check that the debugger does not depend on these methods?

Reply:
I don't see anything obvious.  @noahfalk is there some secret way dbi, dac or even the VS debugger can rely on these functions?

Reply:
The task dependencies come directly from VS diagnostic tools and come in three variants that I recall:
a) They listen to ETW events (probably not relevant here)
b) They put breakpoints on specific APIs and inspect the arguments to the functions at the time they are called.
c) They use func-eval and/or IL interpretation to evaluate private functions

@cshung made a list of all these dependencies in the past (though it may have been specific to .Net Native?) and @gregg-miskelly can probably make sure the right folks on VS debugger look at this

Reply:
CC @r-ramesh

Reply:
Yes, this is used by the VS Tasks Window.

Reply:
I added back the two Task methods and made sure ILLink won't remove them.

Reply:
@noahfalk, @gregg-miskelly, @cshung, could we add comments to the code where there's a reflection-based dependency from VS on these internals/privates?  It'll help to avoid issues with such refactorings/deletions in the future.

Reply:
I think adding such comments is definitely a best practice, though there are likely plenty of occurrences where the debugger took a dependency in the past and no comment was added. I created #16788 to address this one. Feel free to merge it or edit it as you see fit

Reply:
You added a comment to GetActiveTaskFromId. Is GetActiveTasks also used by VS debugger?

Reply:
I can't find any calls of 'GetActiveTasks'.

Reply:
I'll let @gregg-miskelly field these.

Reply:
I answered one minute before you :)

Reply:
Ok, I re-applied the change to remove GetActiveTasks.